algorithm/cgal/FromPolygonSoup.h

```diff
@@ -22,32 +28,32 @@ int FromPolygonSoup(Geometry* geometry, emscripten::val fill) {
   CGAL::Polygon_mesh_processing::polygon_soup_to_polygon_mesh(points, polygons,
                                                               mesh);
 
-  bool failed = false;
-  while (!failed && !CGAL::is_closed(mesh)) {
-    for (const Surface_mesh::Halfedge_index edge : mesh.halfedges()) {
-      if (mesh.is_border(edge)) {
-        std::vector<Face_index> faces;
-        CGAL::Polygon_mesh_processing::triangulate_hole(
-            mesh, edge, std::back_inserter(faces),
-            CGAL::parameters::use_2d_constrained_delaunay_triangulation(false));
-        if (faces.empty()) {
-          failed = true;
-        }
-        break;
-      }
+  if (wrap_always || CGAL::Polygon_mesh_processing::does_self_intersect(mesh)) {
```

Comment:
How fast is wrapping?

Reply:
On my machine, it's tens of seconds for most shapes, which isn't fast, but is usable.